### PR TITLE
call Fetch to extract byte stream and content type

### LINKED ISSUE #10: Clarify interaction with CORS (preflight requests)
From an older thread: https://lists.w3.org/Archives/Public/public-web-perf/2014Jan/0003.html ...

> When using beacon against a cross origin, is it clear that we might need to do a preflight for content types that are not specified by CORS?  For example, if you send a ArrayBuffer, we are probably going to need to preflight.  This is clear when reading the CORS spec, but it isn’t obviously clear from the beacon spec that there might be two http requests for a beacon — one for the OPTIONS and one for the POST.  I am not sure if we can or should relax this or not.  Maybe we should be more explicit about which mime types in the beacon specification don’t cause the OPTIONS request.

Comment from @bzbarsky: https://lists.w3.org/Archives/Public/public-web-perf/2014Jan/0008.html ...

> 1) The way Beacon currently invokes "make a cross-origin request" is weird: it doesn't pass the parameters that operation needs.
> 
> 2) It's not clear to me what happens when "mime type" ends up null at the end of step 7 in the processing model.  Nothing really seems to define the behavior in this situation; I suspect that in practice step 7 should ensure that this situation never happens (that is, that a mime type is always defined by the time that step finishes).

---

1. `data` parameter is [allowed to be one](http://w3c.github.io/beacon/#sec-sendBeacon-method) of ArrayBufferView, Blob, DOMString, or FormData. In our [Processing Model](http://w3c.github.io/beacon/#sec-processing-model), we should make sure we always set the Content-Type... 
  * If ArrayBuffer is being sent, what should we set the `Content-Type` to?
  * If Blob is being sent, and `type` attribute is omitted, what should we set the `Content-Type` to?

1. In [step 10](http://w3c.github.io/beacon/#sec-processing-model) we set the Origin header, mode to CORS, and set the "include credentials" flag, prior to invoking Fetch in step 11. 
  * Is there anything else we need or should do in Beacon spec to trigger proper CORS processing in Fetch? /cc @annevk.

### PR DESCRIPTION
**Pull request by igrigorik**
Current definition replicates Fetch, which is both unnecessary and error prone.

Closes #10.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -304,13 +303,13 @@
       <h2><code>sendBeacon</code> Method</h2>
       <pre class="idl">
 partial interface <dfn id="Navigator">Navigator</dfn> {
-    boolean <a>sendBeacon</a>(<a>DOMString</a> <var title=
-"url">url</var>, optional (<a>ArrayBufferView</a> or <a>Blob</a> or <a>DOMString</a> or <a>FormData</a>)? <var>data</var> = null);
+    boolean <a>sendBeacon</a>(<a>USVString</a> <var title=
+"url">url</var>, optional <a>BodyInit</a>? <var>data</var> = null);
```
Comment (annevk) [inline index.html]: I guess there's no way to get rid of `= null` at this point? Omitting the argument (or using undefined, which is identical) would be much cleaner.

File: index.html [hunk 2]
```diff
@@ -391,49 +389,16 @@ <h4 id="return-values">Return Value</h4>
...
-              by the <a><code>multipart/form-data</code> encoding
-              algorithm</a>.</p>
-            </dd>
-          </dl>
+        <li>Otherwise, create the following temporary variables:
+          <ul>
+            <li>
+              <a href=
+              "https://fetch.spec.whatwg.org/#concept-bodyinit-extract">Extract</a>
+              object's byte stream (<var>transmittedData</var>) and mime type
```
Comment (annevk) [inline index.html]: MIME*

Comment (igrigorik) [inline index.html]: fixed

File: index.html (hunk 1 above)
Comment (igrigorik) [inline index.html]: Yeah, I think that's a separate discussion.

**Comment by igrigorik** [comment]
@annevk any other feedback to address? 
@plehegar @toddreifsteck same thing, any thoughts on this one?

**Comment by annevk** [comment]
I have nothing else on this PR. I might have some stuff on the main specification though, not sure. E.g., it should be setting https://fetch.spec.whatwg.org/#keep-alive-flag to survive fetch groups.

**Comment by toddreifsteck** [comment]
LGTM, Merging

**Pull request merged**